src/Serilog/Formatting/Display/MessageTemplateTextFormatter.cs

```diff
@@ -91,6 +91,15 @@ public void Format(LogEvent logEvent, TextWriter output)
 
                     token.Render(overridden, output, _formatProvider);
                 }
+                else if (pt.IsLevelProperty)
```

Comment:
(In general, allocating objects is something to avoid in the logging pipeline, `Dictionary<,>` turns out to be particularly expensive.)